lib/plugins/package/lib/zipService.js

```diff
@@ -131,53 +144,48 @@ function excludeNodeDevDependencies(servicePath) {
     });
 
     // filter out non node_modules file paths
-    const relevantFilePaths = _.filter(packageJsonFilePaths, (filePath) => {
+    const packageJsonPaths = _.filter(packageJsonFilePaths, (filePath) => {
       const isNodeModulesDir = !!filePath.match(/node_modules/);
       return !isNodeModulesDir;
     });
 
-    _.forEach(relevantFilePaths, (relevantFilePath) => {
-      // the path where the package.json file lives
-      const fullPath = path.join(servicePath, relevantFilePath);
-      const rootDirPath = fullPath.replace(path.join(path.sep, 'package.json'), '');
-
-      process.chdir(rootDirPath);
-
-      // TODO replace with package-manager independent directory traversal?!
-      const prodDependencies = childProcess
-        .execSync('npm ls --prod=true --parseable=true --silent')
-        .toString().trim();
+    if (_.isEmpty(packageJsonPaths)) {
+      return BbPromise.resolve(exAndIn);
+    }
 
+    // NOTE: using mapSeries here for a sequential computation (w/o race conditions)
+    return BbPromise.mapSeries(packageJsonPaths, (packageJsonPath) => {
+      // the path where the package.json file lives
+      const fullPath = path.join(servicePath, packageJsonPath);
+      const dirWithPackageJson = fullPath.replace(path.join(path.sep, 'package.json'), '');
+
+      // we added a catch which resolves so that npm commands with an exit code of 1
+      // (e.g. if the package.json is invalid) won't crash the dev dependency exclusion process
+      return childProcess.execAsync(
+        `npm ls --dev=true --parseable=true --silent >> ${nodeDevDepFile}`,
+        { cwd: dirWithPackageJson }
+      ).catch(() => BbPromise.resolve());
+    })
+    .then(() => fs.readFileAsync(nodeDevDepFile))
+    .then((fileContent) => {
+      const dependencies = _.filter(
```

Comment:
Could you use _.compact here instead of _.filter?
https://lodash.com/docs/4.17.4#compact